editor/editor_export_godot3.cpp

```diff
@@ -1980,7 +1981,190 @@ void EditorExportGodot3::_save_config(const String &p_path) {
+	ERR_FAIL_COND_V(!dst.operator->(), FAILED);
+
+	String http_var = "";
+
+	while (!src->eof_reached()) {
+		String line = src->get_line();
+
+		// Convert _fixed_process( => _physics_process(
+		RegEx regexp("(.*)_fixed_process\\((.*)");
+		int res = regexp.find(line);
+		if (res >= 0 && regexp.get_capture_count() == 3) {
+			line = regexp.get_capture(1) + "_physics_process(" + regexp.get_capture(2);
+		}
+		regexp.clear();
+
+		// Convert RawArray() => PoolByteArray()
+		regexp.compile("(.*)RawArray\\(\\)(.*)");
+		res = regexp.find(line);
+		if (res >= 0 && regexp.get_capture_count() == 3) {
+			line = regexp.get_capture(1) + "PoolByteArray()" + regexp.get_capture(2);
+		}
+		regexp.clear();
+
+		// Convert Vector2Array() => PoolVector2Array()
+		regexp.compile("(.*)Vector2Array\\(\\)(.*)");
+		res = regexp.find(line);
+		if (res >= 0 && regexp.get_capture_count() == 3) {
+			line = regexp.get_capture(1) + "PoolVector2Array()" + regexp.get_capture(2);
+		}
+		regexp.clear();
+
+		// Convert ReferenceFrame => Control
+		regexp.compile("(.*)ReferenceFrame(.*)");
+		res = regexp.find(line);
+		if (res >= 0 && regexp.get_capture_count() == 3) {
+			line = regexp.get_capture(1) + "Control" + regexp.get_capture(2);
+		}
+		regexp.clear();
+
+		// Try to detect a HTTPClient object
+		regexp.compile("[ \t]*([a-zA-Z0-9_]*)[ ]*=[ ]*HTTPClient\\.new\\(\\)");
+		res = regexp.find(line);
+		if (res >= 0 && regexp.get_capture_count() == 2) {
+			http_var = regexp.get_capture(1).strip_edges();
+		}
+		regexp.clear();
+
+		if (http_var != "") {
+			// Convert .connect( => .connect_to_host(
+			regexp.compile("(.*)" + http_var + "\\.connect\\((.*)");
+			res = regexp.find(line);
+			if (res >= 0 && regexp.get_capture_count() == 3) {
+				line = regexp.get_capture(1) + http_var + ".connect_to_host(" + regexp.get_capture(2);
+			}
+			regexp.clear();
+		}
+
+		int count;
+		int tries = 0;
+		do {
+			count = 0;
+
+			// Convert _pos( => _position(
+			regexp.compile("(.*)_pos\\((.*)");
+			res = regexp.find(line);
+			if (res >= 0 && regexp.get_capture_count() == 3) {
+				line = regexp.get_capture(1) + "_position(" + regexp.get_capture(2);
+				count++;
+			}
+			regexp.clear();
+
+			// Convert KEY_RETURN => KEY_ENTER
+			regexp.compile("(.*)KEY_RETURN(.*)");
+			res = regexp.find(line);
+			if (res >= 0 && regexp.get_capture_count() == 3) {
+				line = regexp.get_capture(1) + "KEY_ENTER" + regexp.get_capture(2);
+				count++;
+			}
+			regexp.clear();
+
+			// Convert .get_opacity() => .get_modulate().a
+			regexp.compile("(.*)\\.get_opacity\\(\\)(.*)");
+			res = regexp.find(line);
+			if (res >= 0 && regexp.get_capture_count() == 3) {
+				line = regexp.get_capture(1) + ".get_modulate().a" + regexp.get_capture(2);
+				count++;
+			}
+			regexp.clear();
+
+			// Convert .set_opacity(var) => .modulate.a = var
+			regexp.compile("([ \t]*)([a-zA-Z0-9_]*)[ ]*\\.set_opacity\\((.*)\\)(.*)");
```

Comment:
@marcelofg55 I'm doing some improvements to the script conversion, and wonder about this regexp. What's the point of `[ ]*` before `\\.set_opacity\\(`? Shouldn't it simply be `"(.*)\\.set_opacity\\((.*)\\)(.*)"`?

Reply:
I don't remember if there was a reason that I wrote that regexp that way, but it should work fine as you say as `(.*)\\.set_opacity\\((.*)\\)(.*)`.